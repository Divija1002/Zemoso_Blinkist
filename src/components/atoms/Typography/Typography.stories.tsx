import { Story } from "@storybook/react";
import TypographyText, { TypographyTextType } from ".";

export default {
  title: "components/atoms/Typography",
  component: TypographyText,
};

const Template: Story<TypographyTextType> = (args) => (
  <TypographyText {...args}>Typography</TypographyText>
);

export const Text = Template.bind({});

Text.args = {
  variant: "h1",
  sx: {},
};
