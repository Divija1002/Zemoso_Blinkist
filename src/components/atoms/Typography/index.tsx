import React from "react";
import { Typography, Theme } from "@mui/material";
import { SxProps } from "@mui/system";

interface TypographyTextType {
  variant?:
    | "h1"
    | "h2"
    | "h3"
    | "h4"
    | "h5"
    | "h6"
    | "subtitle1"
    | "subtitle2"
    | "body1"
    | "body2"
    | "caption"
    | "button"
    | "overline"
    | "inherit";
  children: React.ReactNode;
  sx?: SxProps<Theme>;
}

const TypographyText = ({ variant, children, sx }: TypographyTextType) => {
  return (
    <Typography variant={variant} sx={sx}>
      {children}
    </Typography>
  );
};

export default TypographyText;
