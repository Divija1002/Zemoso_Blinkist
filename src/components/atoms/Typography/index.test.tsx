import renderer from "react-test-renderer";
import Typography from ".";

describe("Typography Snapshot", () => {
  it("Typography Testing", () => {
    const renderedComponent = renderer.create(<Typography>Hello</Typography>);
    expect(renderedComponent).toMatchSnapshot();
  });
});
