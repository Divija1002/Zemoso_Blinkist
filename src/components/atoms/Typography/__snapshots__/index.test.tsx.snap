// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Snapshot Typography Testing 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
>
  Hello
</p>
`;
